crates/polars-ops/src/chunked_array/binary/namespace.rs

```diff
@@ -6,26 +6,40 @@ use base64::engine::general_purpose;
 #[cfg(feature = "binary_encoding")]
 use base64::Engine as _;
 use memchr::memmem::find;
+use polars_core::prelude::arity::binary_elementwise;
 
 use super::*;
 
 pub trait BinaryNameSpaceImpl: AsBinary {
     /// Check if binary contains given literal
-    fn contains(&self, lit: &[u8]) -> PolarsResult<BooleanChunked> {
+    fn contains(&self, lit: &[u8]) -> BooleanChunked {
         let ca = self.as_binary();
         let f = |s: &[u8]| find(s, lit).is_some();
-        let mut out: BooleanChunked = if !ca.has_validity() {
-            ca.into_no_null_iter().map(f).collect()
+        if !ca.has_validity() {
+            ca.into_no_null_iter()
+                .map(f)
+                .collect::<BooleanChunked>()
+                .with_name(ca.name())
         } else {
-            ca.into_iter().map(|opt_s| opt_s.map(f)).collect()
-        };
-        out.rename(ca.name());
-        Ok(out)
+            ca.into_iter()
+                .map(|opt_s| opt_s.map_or(false, f))
+                .collect::<BooleanChunked>()
+                .with_name(ca.name())
+        }
     }
 
-    /// Check if strings contain a given literal
-    fn contains_literal(&self, lit: &[u8]) -> PolarsResult<BooleanChunked> {
-        self.contains(lit)
+    fn contains_chunked(&self, lit: &BinaryChunked) -> BooleanChunked {
+        let ca = self.as_binary();
+        match lit.len() {
+            1 => match lit.get(0) {
+                Some(pat) => ca.contains(pat),
+                None => BooleanChunked::full(ca.name(), false, ca.len()),
+            },
+            _ => binary_elementwise(ca, lit, |opt_src, opt_lit| match (opt_src, opt_lit) {
+                (Some(src), Some(lit)) => Some(find(src, lit).is_some()),
+                _ => Some(false),
```

Comment:
We should propagate nulls here.

Reply:
If `lit` is null we should still propagate null. It was a bug if we didn't do this.

Reply:
Because null values are missing. We cannot answer the question if we the value is in the data, as there could be any data behind that missing value. So we propagate nulls. So the output of `l` should be `null, true, null`.

Reply:
Yes, that follows the same rationale.